### PR TITLE
[SC-430] Revert on non-existent contract instead of silent success

### PR DESCRIPTION
**Pull request by hexonaut**
Switched to using the OZ `Address` helper library which does an additional check for empty addresses and reverts if so. This will help with things like cross-chain payload e2e testing to have them fail when the address is not deployed. Also added some more testing coverage for these types of failure cases with delegatecall and call.

### CODE REVIEW

**Review by barrutko — DISMISSED** [review]

File: test/AuthBridgeExecutor.t.sol [hunk 1]
```diff
@@ -395,6 +395,54 @@ contract AuthBridgeExecutorExecuteTests is AuthBridgeExecutorTestBase {
         executor.execute(0);
     }
 
+    function test_execute_delegateCallEmptyContract() public {
+        // Trigger some evm error like trying to call a non-payable function
```
Comment (lucas-manuel) [inline test/AuthBridgeExecutor.t.sol]: Can remove comment

File: test/AuthBridgeExecutor.t.sol [hunk 2]
```diff
@@ -395,6 +395,54 @@ contract AuthBridgeExecutorExecuteTests is AuthBridgeExecutorTestBase {
...
+        _queueAction(action);
+        skip(DELAY);
+
+        // Should return the underlying error message
+        vm.expectRevert("An error occurred");
+        executor.execute(0);
+    }
+
+    function test_execute_callEmptyContract() public {
+        // Trigger some evm error like trying to call a non-payable function
```
Comment (lucas-manuel) [inline test/AuthBridgeExecutor.t.sol]: Remove comment

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/AuthBridgeExecutor.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/AuthBridgeExecutor.t.sol]: fixed

File: test/AuthBridgeExecutor.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/AuthBridgeExecutor.t.sol]: fixed

**Review by barrutko — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**